xrmocap/ops/bottom_up_association/bottom_up_associator.py

```diff
@@ -0,0 +1,169 @@
+# yapf: disable
+import logging
+import numpy as np
+from typing import List, Mapping, Tuple, Union
+import json
+import cv2
+from xrprimer.data_structure.camera import (
+    FisheyeCameraParameter, PinholeCameraParameter,
+)
+from xrprimer.utils.log_utils import get_logger
+
+from xrmocap.data_structure.keypoints import Keypoints
+
+from xrmocap.ops.triangulation.builder import (
+    BaseTriangulator, build_triangulator,
+)
+
+from xrmocap.ops.bottom_up_association.matching.builder import build_matching
+# yapf: enable
+
+
+class Camera():
+    def __init__(self,cam_param) -> None:
+        super().__init__()
+        self.K = cam_param.intrinsic33()
+        self.T = np.array(cam_param.get_extrinsic_t())
+        self.R = np.array(cam_param.get_extrinsic_r())
+        self.Ki = np.linalg.inv(self.K)
+        self.Rt = self.R.T
+        self.RtKi = np.matmul(self.Rt, self.Ki)
+        self.Pos = -np.matmul(self.Rt, self.T) 
+        
+        self.Proj = np.zeros((3,4), dtype=np.float)
+        for i in range(3):
+            for j in range(4):
+                self.Proj[i,j] = self.R[i,j] if j < 3 else self.T[i]
+        self.Proj = np.matmul(self.K, self.Proj)
+
+    def calcRay(self, uv):
+        ver  = -self.RtKi.dot(np.append(uv, 1).T)
+        return ver / np.linalg.norm(ver)
+
+class BottomUpAssociator:
+
+    def __init__(self,
+                 triangulator: Union[None, dict, BaseTriangulator],
+                 m_filter: bool=False,
+                 fourd_matching: Union[None, dict] = None,
+                 min_asgn_cnt: int = 5 ,
+                 logger: Union[None, str, logging.Logger] = None) -> None:
+
+        self.logger = get_logger(logger)
+
+        if isinstance(triangulator, dict):
+            triangulator['logger'] = self.logger
+            self.triangulator = build_triangulator(triangulator)
+        else:
+            self.triangulator = triangulator
+        
+        self.n_views = -1
+        self.last_multi_kps3d = dict()
+        self.joint_size = 19
```

Comment:
In our code, we use kps3d instead of joint (If it represents a two-dimensional joint, we use kps2d), and n_* to denote the number of a noun, so rename `self.joint_size` to `self.n_kps3d`.

Reply:
I will resolve